pkg/git/changelog.go

```diff
@@ -0,0 +1,267 @@
+		}
+
+		gitChangesHolder[index] = change
+
+	}
+
+	changes := GitChanges(gitChangesHolder)
+	versionBump := changes.GetVersionBump()
+	changeLog := GitChangeLog{
+		VersionBump: versionBump,
+		Changes:     changes,
+		OutputMessage: GetChangeLogOutputMessage(versionBump, changes, options),
+	}
+
+	return changeLog, err
+}
+
+func CommitIsValid(title string) bool {
+	valid, _ := regexp.MatchString(`.*(\()(.)*(\))(:){1}`, title)
+	return valid
+}
+
+func CommitHasBreakingChange(body string) bool {
+	var breakingChangeFound = true
+	breakingChangeFormat := regexp.MustCompile(`BREAKING CHANGE: .*\n`)
+	breakingChange := breakingChangeFormat.FindAllString(body, -1)
+
+	if len(breakingChange) == 0 {
+		breakingChangeFound = false
+	}
+	return breakingChangeFound
+}
+
+func (g GitChanges) GetVersionBump() string {
+	majorChanges, minorChanges, patchChanges, _ := g.GetSeparatedChanges()
+
+	if len(majorChanges) > 0 {
+		return "MAJOR"
+	} else if len(minorChanges) > 0 {
+		return "MINOR"
+	} else if len(patchChanges) > 0 {
+		return "PATCHES"
+	}
+	return "NONE"
+}
+
+func (g GitChanges) GetSeparatedChanges() ([]GitChange, []GitChange, []GitChange, []GitChange) {
+	majorChanges := make([]GitChange, 0)
+	minorChanges := make([]GitChange, 0)
+	patchChanges := make([]GitChange, 0)
+	unknownChanges := make([]GitChange, 0)
+
+	for _, change := range g {
+		if change.BreakingChange {
+			majorChanges = append(majorChanges, change)
+		} else if _, ok := minorTypes[change.CommitType]; ok {
+			minorChanges = append(minorChanges, change)
+		} else if _, ok := patchTypes[change.CommitType]; ok {
+			patchChanges = append(patchChanges, change)
+		} else {
+			unknownChanges = append(unknownChanges, change)
+		}
+	}
+
+	return majorChanges, minorChanges, patchChanges, unknownChanges
+}
+
+func GetStoryLink(svc issues.IssueService, owner string, repo string, issue string) (string, error) {
+	issueNum, err := strconv.Atoi(issue)
+	if err != nil {
+		return "", err
+	}
+	iss, err := svc.GetParents(issues.NewIssueRef(owner, repo, issueNum))
+	if err != nil {
+		return "", err
+	}
+	if len(iss) != 2 {
+		// this means the issue number must be wrong
+		return "", err
+	}
+	storyLink := new(strings.Builder)
+	fmt.Fprintf(storyLink, "https://github.com/")
+	fmt.Fprintf(storyLink, iss[1].Org)
+	fmt.Fprintf(storyLink, "/")
+	fmt.Fprintf(storyLink, iss[1].Repo)
+	fmt.Fprintf(storyLink, "/issues/")
+	fmt.Fprintf(storyLink, strconv.Itoa(iss[1].Number))
+
+	return storyLink.String(), err
+}
+
+func GetIssueLink(dir string, issue string) string {
+	linkFormat := regexp.MustCompile(`github\.com.*`)
+	issueLink := new(strings.Builder)
+	fmt.Fprintf(issueLink, "https://")
+	fmt.Fprintf(issueLink, linkFormat.FindString(dir))
+	fmt.Fprintf(issueLink, "/issues/")
+	fmt.Fprintf(issueLink, issue)
+
+	return issueLink.String()
+}
+
+func GetCommitTitle(lines []string) string {
+	return lines[4]
+}
+
+func GetRepo(dir string) string {
+	return strings.Split(dir, "/")[6]
```

Comment:
What happens if the repo is cloned to `/root/src/naveegoinc/bosun`?

Check out the `GetOrgAndRepoFromPath` function in this package.